vumi/utils.py

```diff
@@ -13,12 +13,14 @@
 from twisted.internet import protocol
 from twisted.internet.defer import succeed
 from twisted.python.failure import Failure
-from twisted.web.client import Agent, ResponseDone, WebClientContextFactory
+from twisted.web.client import Agent, ResponseDone
 from twisted.web.server import Site
 from twisted.web.http_headers import Headers
 from twisted.web.iweb import IBodyProducer
 from twisted.web.http import PotentialDataLoss
 from twisted.web.resource import Resource
+from treq._utils import default_pool, default_reactor
```

Comment:
when calling `default_pool` we were sending `pool=None` and `persistence=False`. In that case the method will create a new HTTPSConnectionPool. so that is what I did instead.
https://github.com/twisted/treq/blob/master/src/treq/_utils.py#L38-L42